src/main.ts

```diff
@@ -48,64 +69,56 @@ async function run() {
         }
       }
 
-      if (context.payload.repository.private !== true) {
-        core.info("Checking diff contents");
-        const diffContains = core.getInput("diffContains");
-        const diffDoesNotContain = core.getInput("diffDoesNotContain");
-        const diff_url = context.payload.pull_request.diff_url;
-        core.info("Requesting " + diff_url);
-        const result = await github.request(diff_url);
-        const files = parse(result.data);
-        core.exportVariable("files", files);
-        core.setOutput("files", files);
-        const filesChanged = +core.getInput("filesChanged");
-        if (filesChanged && files.length != filesChanged) {
-          core.setFailed(
-            "You should change exactly " + filesChanged + " file(s)"
-          );
-        }
+      core.info("Checking diff contents");
+      const diffContains = core.getInput("diffContains");
+      const diffDoesNotContain = core.getInput("diffDoesNotContain");
+
+      const files = await getDiff(octokit, context);
```

Comment:
So what you've done here is factor out diff processing to the function. Good one.